### ISSUE TITLE
Vector fields

### ISSUE THREAD

**Issue by Benny121221** [swharden/ScottPlot#438]
**What can we do to make ScottPlot better?**
Add a vector field plottable, e.g.: 
![image](https://user-images.githubusercontent.com/8635304/83201571-04756e00-a103-11ea-9e46-d3ce3452ef81.png)

Does your proposed feature allow users to do new things in source code? If so, show us how you'd like to be able to use the new feature:

```cs
plt.PlotVectorField(Vector2[] vectors, double[] xs, double[] ys)
```

It might be valuable for the arrows to change in colour to indicate magnitude, as vector fields typically have scaled-down arrows so the plot remains readable

Conveniently, Microsoft provides vector types for 2, 3, and 4 dimensions: https://docs.microsoft.com/en-us/dotnet/api/system.numerics.vector2?view=netcore-3.1

**Comment by swharden** [comment]
Haha, consistent with previous suggestions @Benny121221 you surprised me with a really cool idea that seemed to come out of nowhere! I like it.

Use of System.Numerics is an excellent idea. I've been getting familiar with their `Complex` struct/methods lately with [FftSharp](https://github.com/swharden/FftSharp)

You may find the `PlotArrow()` method useful to refer to if you decide to make a demo of this one.

https://github.com/swharden/ScottPlot/blob/641126646d77c3aefad055e9f31d44da333eb03f/src/ScottPlot/Plot.cs#L689-L718

**Comment by swharden** [comment]
We have some experimental code merged in the master branch now (#424)

There may be some utility in expanding the colormap module to let those colormaps also be used to color line plots to produce graphs like this:

![image](https://user-images.githubusercontent.com/4165489/83372829-4f99b600-a37b-11ea-8116-93889cbbbcd1.png)

**Comment by swharden** [comment]
Just merged this in, thanks @Benny121221!
